product_attribute_company_favorite/models/product_attribute_is_favorite_mixin.py

```diff
@@ -0,0 +1,43 @@
+from odoo import api, fields, models
+
+
+class ProductAttributeIsFavoriteMixin(models.AbstractModel):
+    _name = "product.attribute.favorite.mixin"
+    _description = """Methods used both in product.attribute and
+    product.attribute.value to implement is_favorite functionalities
+    """
+
+    is_favorite = fields.Boolean(
+        string="Favorite",
+        company_dependent=True,
+        default=True,
+        help="""If checked, this record will appear when searched
+        in product attribute list in the product form""",
+    )
+
+    def _name_search(
+        self, name, args=None, operator="ilike", limit=100, name_get_uid=None
+    ):
+        args = list(args or [])
+        args += [("is_favorite", "=", True)]
+        return super()._name_search(
+            name, args=args, operator=operator, limit=limit, name_get_uid=name_get_uid
+        )
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        new_record_favorite_for_all_companies = (
+            self.env["ir.config_parameter"]
+            .sudo()
+            .get_param(self.CREATION_ACROSS_COMPANY_CONFIG)
+        )
+        records = super().create(vals_list)
+        if new_record_favorite_for_all_companies:
+            company_ids = (
+                self.env["res.company"].with_context(active_test=False).search([]).ids
+            )
+            for record in records:
+                for company_id in company_ids:
+                    contextual_record = record.with_company(company_id)
+                    contextual_record.is_favorite = True
```

Comment:
There is a potential bug here, although a little unlikely. If `vals_list` contains `{"is_favorite": False}` for ≥1 record, that instruction will be overwritten by the code below.

Instead of writing `True` to the field, you should probably write `record.is_favorite`.